spec/multi_patient_api/bulk_export_validation_tester_spec.rb

```diff
@@ -0,0 +1,72 @@
+  end
+  let(:generic_block) { proc { |chunk| } }
+  let(:streamed_chunks) { [] }
+  let(:streamed_response) { [] }
+  let(:process_line_block) { proc { |chunk| streamed_chunks << ("#{chunk} touched") } }
+  let(:process_response_block) { proc { |response| streamed_response << response[:headers][0].value } }
+  let(:client) do
+    block = proc { url url }
+    Inferno::DSL::HTTPClientBuilder.new.build(group, block)
+  end
+
+  describe 'stream_ndjson' do
+    it 'makes a stream request using the given endpoint' do
+      stub_request(:get, url.to_s)
+        .to_return(status: 200, body: '', headers: {})
+
+      group.stream_ndjson(url, {}, generic_block, generic_block)
+
+      expect(group.response[:status]).to eq(200)
+    end
+
+    it 'applies process_header once, upon reception of a single, one-line chunk' do
+      stub_request(:get, url.to_s)
+        .to_return(status: 200, body: basic_body, headers: { 'Application' => 'expected' })
+
+      group.stream_ndjson(url, {}, generic_block, process_response_block)
+
+      expect(streamed_response).to eq(['expected'])
+    end
+
+    it 'applies process_header several times, upon reception of multiple chunks' do
```

Comment:
Should this test be implemented or removed?

Reply:
Removing it, unfortunately. The Faraday streaming capability is not great, and only gives access to the piecewise streamed-in chunks -- meaning I can't think of a way to check the response headers until after the full response is streamed in. So, can only check headers once after full response is received, which is less ideal than if we knew off the bat that say the content-type was wrong.